pkg/app/piped/platformprovider/kubernetes/manifest.go

```diff
@@ -224,12 +224,17 @@ func ParseManifests(data string) ([]Manifest, error) {
 		manifests = make([]Manifest, 0, len(parts))
 	)
 
-	for _, part := range parts {
+	newLineAtEOF := strings.HasSuffix(data, "\n")
+	for i, part := range parts {
 		// Ignore all the cases where no content between separator.
 		part = strings.TrimSpace(part)
```

Comment:
Looks like in case of the yaml file contains more than one empty line at the end of file, only one empty line is remained thus the diff will be shown, isn't it? In that case, lets add test to demonstrate that and I will merge this PR 🙌

Reply:
Thanks for pointing that out. I have fixed it and added a test case. https://github.com/pipe-cd/pipecd/pull/4033/commits/0e250b194b6ead5508f41e4a9eb65ef445bfe3c1